bundles/framework/language-selector/components/LanguageSelector.jsx

```diff
@@ -0,0 +1,37 @@
+import React from 'react';
+import styled from 'styled-components';
+import {languageNames} from './languageNames';
+import {Select, Option} from '../../../admin/admin-layereditor/components/Select';
+import {getSupportedLanguages} from './getSupportedLanguages';
+
+const StyledSelect = styled(Select)`width: 100%;`;
+
+const loc = Oskari.getMsg.bind(null, 'language-selector');
+
+const handleLanguageChange = (language) => {
+    let redirect;
+    const newLang = 'lang=' + language;
+    if (window.location.href.indexOf('?') === -1) {
+        redirect = window.location.href + '?' + newLang;
+    } else if (window.location.href.indexOf('lang=') > -1) {
+        redirect = window.location.href.replace(/lang=[^&]+/, newLang);
+    } else {
+        redirect = window.location.href + '&' + newLang;
+    }
+    window.location.href = redirect;
+};
+
+export const LanguageSelector = () => {
+    const supported = getSupportedLanguages();
```

Comment:
Should the supported and current languages be passed as props instead? The only localization requires knowledge of Oskari? Maybe the change handle callback as well? Then this could be used to do any language selection in Oskari. For example some admin for selecting the default instance language etc.

Reply:
Maybe even pass the two strings needed from localization so the component could be "Oskari-free" :)